src/components/AmountFilter.jsx

```diff
@@ -58,12 +61,20 @@ export default function AmountFilter() {
           <Grid item className={classes.filterStyle}>
             <InputLabel className={classes.labelStyle}>Max Amount</InputLabel>
             <AmountTextField
-              value={values.max}
-              onChange={(e) => handleChange('max', e.target.value)}
+              value={maxAmount || ''}
+              onChange={(e) =>
+                onAmountFilterChange('maxAmount', e.target.value || 0)
+              }
             />
           </Grid>
         </Grid>
       </Popover>
     </>
   );
 }
+
+AmountFilter.propTypes = {
+  minAmount: PropTypes.number.isRequired,
+  maxAmount: PropTypes.number.isRequired,
```

Comment:
I think the shorter names `min` and `max` are straightforward for this component.